core/ui/EditorToolbar/picture-dropdown.tid

```diff
@@ -4,8 +4,44 @@ title: $:/core/ui/EditorToolbar/picture-dropdown
 [img[$(imageTitle)$]]
 \end
 
+\define insert-external-image()
```

Comment:
No problem. As mentioned before, I am not a coder by any stretch of the word, just a bare copy-paster at tops. I hacked this together, but like you said this could be tremendously improved if input text in the field would:

1. Filter the displayed results for image tiddlers.

2. Ideally nothing would show up initially until the user typed something, since if there are lots of internal images, loading them all for displaying is quite slow and locks up the browser.

3. This would allow inserting external images with an protocol+UTL (like http://path.to/image/file.jpg), or relative path (like media/path/to/image.jpg)

3. For external images tiddlywiki would try to display a small preview/thumbnail in the dropdown of the resulting path (as you suggested) confirming it was valid

2. Like the proposal I did before for the links dropdown this would also allow inserting not-yet existing image tiddlers

4. Pressing enter would commit whatever was introduced into the text box, internal image tiddler or not

Well at least that is what I would do, if I had the skills to accomplish it ;) I was in the hopes that Jeremy, or another talented developer such as yourself would improve it based on this concept or starting from scratch.